frontend/src/components/common/Resource/EditorDialog.tsx

```diff
@@ -557,3 +587,263 @@ export default function EditorDialog(props: EditorDialogProps) {
+          sx={theme => ({
+            position: 'relative',
+            borderRadius: 2,
+            p: 4,
+            textAlign: 'center',
+            border: '2px dashed',
+            borderColor: dragOver
+              ? 'primary.main'
+              : theme.palette.mode === 'dark'
+              ? 'grey.300'
+              : 'grey.800',
+            bgcolor: theme.palette.mode === 'dark' ? 'grey.800' : 'grey.300',
+            cursor: 'pointer',
+            transition: '0.2s',
+          })}
+        >
+          <input
+            type="file"
+            style={{
+              position: 'absolute',
+              top: 0,
+              left: 0,
+              width: '100%',
+              height: '100%',
+              opacity: 0,
+              cursor: 'pointer',
+            }}
+            onChange={e => setUploadedFile(e.target.files?.[0] || null)}
+          />
+
+          <Icon icon="mdi:upload" width="40" color="white" />
+          <Typography sx={{ m: 2 }}>
+            {dragOver
+              ? t('translation|Drop the file here...')
+              : t('translation|Select a file or drag and drop here')}
+          </Typography>
+          <Box
+            sx={{
+              p: 1,
+              bgcolor: 'yellow',
+              borderRadius: 1,
+              color: 'black',
+              fontWeight: 'bold',
+              '&:hover': { bgcolor: '#e6d600' },
+            }}
+            component="span"
+          >
+            {t('translation | Select File')}
+          </Box>
+        </Box>
+
+        {uploadedfile && (
+          <Box
+            sx={theme => ({
+              bgcolor: theme.palette.mode === 'dark' ? 'grey.800' : 'grey.300',
+              color: theme.palette.mode === 'dark' ? 'grey.300' : 'grey.800',
+              borderRadius: 1,
+              mt: 2,
+              p: 1,
+              width: '100%',
+              border: '1px',
+              fontWeight: 'bold',
+            })}
+          >
+            {uploadedfile.name}
+          </Box>
+        )}
+        <Box sx={{ display: 'flex', mt: 2, justifyContent: 'space-between' }}>
+          <Button
+            variant="outlined"
+            sx={{ borderColor: 'grey.500', px: 2 }}
+            onClick={() => setUploadYaml(false)}
+          >
+            <Icon
+              icon="mdi:arrow-left"
+              width="18"
+              height="18"
+              style={{ display: 'inline-block', marginRight: '8px' }}
+            />
+            {t('translation|Back')}
+          </Button>
+          <Button
+            onClick={handleLoadFile}
+            variant="contained"
+            sx={{
+              bgcolor: 'grey.700',
+              color: 'white',
+              px: 5,
+              '&:hover': { bgcolor: 'grey.600' },
+            }}
+          >
+            {t('translation|Load')}
+          </Button>
+        </Box>
+      </>
+    );
+  };
+
+  const UploadyamlfromUrl = () => {
```

Comment:
this should be in camel case
```suggestion
  const uploadYamlFromUrl = () => {
```